dogu/doguName.go

```diff
@@ -0,0 +1,78 @@
+package dogu
+
+import (
+	"errors"
+	"fmt"
+	"github.com/cloudogu/cesapp-lib/core"
+	"strings"
+)
+
+type SimpleName string
+type Namespace string
+
+// String returns the string representation of the SimpleName.
+func (s SimpleName) String() string {
+	return string(s)
+}
+
+type QualifiedVersion struct {
+	Name    QualifiedName
+	Version core.Version
+}
+type QualifiedName struct {
+	SimpleName SimpleName
+	Namespace  Namespace
+}
+
+func NewQualifiedName(namespace Namespace, simpleName SimpleName) (QualifiedName, error) {
+	doguName := QualifiedName{Namespace: namespace, SimpleName: simpleName}
+	err := doguName.Validate()
+	if err != nil {
+		return QualifiedName{}, err
+	}
+	return doguName, nil
+}
+
+func NewQualifiedVersion(name QualifiedName, version core.Version) (QualifiedVersion, error) {
+	err := name.Validate()
+	if err != nil {
+		return QualifiedVersion{}, err
+	}
+	return QualifiedVersion{
+		Name:    name,
+		Version: version,
+	}, nil
+}
+
+func (name QualifiedName) Validate() error {
+	var errorList []error
+	if name.Namespace == "" {
+		errorList = append(errorList, fmt.Errorf("namespace of dogu %q must not be empty", name.SimpleName))
+	}
+	if name.SimpleName == "" {
+		errorList = append(errorList, fmt.Errorf("dogu name must not be empty"))
+	}
+	splitName := strings.Split(string(name.Namespace), "/")
+	if len(splitName) > 1 {
```

Comment:
I feel like this is more readable, isn't it?
```suggestion
	if strings.Contains(string(name.Namespace), "/") {
```